code/game/objects/items/kirby_plants/kirbyplants.dm

```diff
@@ -136,8 +136,30 @@
 	name = "Potty the Potted Plant"
 	desc = "A secret agent staffed in the station's bar to protect the mystical cakehat."
 	icon_state = "potty"
+	base_icon_state = "potty"
 	custom_plant_name = TRUE
 	trimmable = FALSE
+	actions_types = list(/datum/action/item_action/toggle_light)
+	action_slots = ALL
+
+	///Boolean on whether the light is on and flashing.
+	var/light_enabled = TRUE
+
+//this is called by the action type as well
+/obj/item/kirbyplants/potty/attack_self(mob/user)
+	. = ..()
+	if(.)
+		return .
+	light_enabled = !light_enabled
+	update_appearance(UPDATE_ICON_STATE)
```

Comment:
```suggestion
	update_appearance()
```
we should just test against this it'll break weirdo overlays

Reply:
wdym? we dont want to update overlays, unless you're saying update appearance cuts overlays even when its not updating them

Reply:
overlays are often dependent on icon state, I can't think of a good reason to ever blacklist them because of this